### ISSUE TITLE
 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)

### ISSUE THREAD

**Issue by ulises2k** [marvinroger/homie-esp8266#740]
Version Homie 3.0.1
ESP8266 - 3.1.1
Arduino IDE 1.8.9

![image](https://user-images.githubusercontent.com/2415609/221435774-9a893ab2-4195-4ec5-b779-78b491d9e560.png)

homie-esp8266-3.0.1\src\Homie\Boot\BootConfig.cpp: In member function 'void HomieInternals::BootConfig::_proxyHttpRequest(AsyncWebServerRequest*)':

homie-esp8266-3.0.1\src\Homie\Boot\BootConfig.cpp:315:20: error: call to 'HTTPClient::begin' declared with attribute error: obsolete API, use ::begin(WiFiClient, url)

**Comment by ulises2k** [comment]
FIX
![image](https://user-images.githubusercontent.com/2415609/221436112-843e344a-92f6-404a-bcaa-f6e07ca3f56f.png)

```
void BootConfig::_proxyHttpRequest(AsyncWebServerRequest *request) {
  Interface::get().getLogger() << F("Received transparent proxy request") << endl;

  String url = String("http://");
  url.concat(request->host());
  url.concat(request->url());

	**WiFiClient client;**
	
  // send request to destination (as in incoming host header)
  _httpClient.setUserAgent(F("ESP8266-Homie"));
  _httpClient.begin(**client**,url);
...

```

**Comment by luebbe** [comment]
This is a duplicate of #711 and it is fixed on the develop branch. I guess there is no official release with this fix

**Comment by trysten** [comment]
Indeed: fixed by #717
I did:
```bash
cd ~/Arduino/libraries
rm -r homie-esp8266-*
git clone https://github.com/homieiot/homie-esp8266.git --depth=1
```
This issue can be closed.
